### PR TITLE
chore(docs): Add security risk warnings to API docs

### LINKED ISSUE #9552: Add annotation to APIs that pose a security risk
Here's a commit that shows how you use the tags. It's not checked in because the text is still awful. In particular, @howToUse is not actually helpful.  https://github.com/naomiblack/angular/commit/b21e6ad9d1b96f3f5c052e059c7bdd58ea348af8

Docs that need this warning are documented in https://docs.google.com/spreadsheets/d/1FsVfgBeaErHrKx4l7sVYwV8BuegyPxPjmspbuThwPuA/edit#gid=406286237
 (internal)

### PR DESCRIPTION
**Pull request by naomiblack**
Martin, have a look please? This is based on the notes in the audit tracking sheet.

closes https://github.com/angular/angular/issues/9552

### CODE REVIEW

File: modules/@angular/compiler/src/runtime_compiler.ts [hunk 1]
```diff
@@ -31,6 +31,11 @@ import {XHR} from './xhr';
  * An internal module of the Angular compiler that begins with component types,
  * extracts templates, and eventually produces a compiled version of the component
  * ready for linking into an application.
+ *
+ * @security  When compiling templates at runtime, you must
```
Comment (mprobst) [inline modules/@angular/compiler/src/runtime_compiler.ts]: nit: align to 100 chars?

File: modules/@angular/core/src/linker/element_ref.ts [hunk 2]
```diff
@@ -12,6 +12,9 @@
  * An `ElementRef` is backed by a render-specific element. In the browser, this is usually a DOM
  * element.
  *
+ * @security Permitting direct access to the DOM can make your application more vulnerable to
+ * XSS attacks. Carefully review any use of `ElementRef` in your code.
```
Comment (mprobst) [inline modules/@angular/core/src/linker/element_ref.ts]: maybe add "Prefer only interacting with the DOM through Angular templates." or something along the lines, i.e. give advice on what to do instead?

File: modules/@angular/http/src/http.ts [hunk 3]
```diff
@@ -202,6 +202,13 @@ export class Jsonp extends Http {
    * a {@link Request} instance. If the first argument is a url, an optional {@link RequestOptions}
    * object can be provided as the 2nd argument. The options object will be merged with the values
    * of {@link BaseRequestOptions} before performing the request.
+   *
+   * @security JSON + CORS is a safer alternative to JSONP for most applications, and is
```
Comment (mprobst) [inline modules/@angular/http/src/http.ts]: Preferred options are (1) regular XHR, if on the same host and then (2) JSON+CORS. People might end up here even though they could use (1) just as well – maybe point them to (1)?

File: modules/@angular/platform-browser/src/security/dom_sanitization_service.ts [hunk 4]
```diff
@@ -83,6 +83,11 @@ export interface SafeResourceUrl extends SafeValue {}
  * does not start with a suspicious protocol, or an HTML snippet that does not contain dangerous
  * code. The sanitizer leaves safe values intact.
  *
+ * @security Calling this API disables Angular's built-in sanitization for the value passed in.
```
Comment (mprobst) [inline modules/@angular/platform-browser/src/security/dom_sanitization_service.ts]: This has to go on the individual `bypassSecurity...` calls, the sanitizer itself is mostly legit (e.g. calling `.sanitize` is fine).

**Comment by mprobst** [comment]
Consider adding links to `http://g.co/ng/security` (or `http://g.co/ng/security#xss`) to the docs strings.

File: modules/@angular/compiler/src/runtime_compiler.ts (hunk 1 above)
Comment (naomiblack) [inline modules/@angular/compiler/src/runtime_compiler.ts]: done

File: modules/@angular/http/src/http.ts (hunk 3 above)
Comment (naomiblack) [inline modules/@angular/http/src/http.ts]: done.

File: modules/@angular/platform-browser/src/security/dom_sanitization_service.ts (hunk 4 above)
Comment (naomiblack) [inline modules/@angular/platform-browser/src/security/dom_sanitization_service.ts]: I can only put @security once for the entire class, but there are warnings on the individual methods. See preview at https://ngdocsdev.firebaseapp.com/docs/ts/latest/api/platform-browser/index/DomSanitizationService-class.html

**Comment by naomiblack** [comment]
Reviewed with Martin. Ready to go.

**Comment by naomiblack** [comment]
squashed and merged as https://github.com/angular/angular/commit/9340e1b065edb66717d937a86098541ac93278c6

**Pull request closed without merging**